.github/workflows/integrationtest.yml

```diff
@@ -0,0 +1,175 @@
+        env:
+          MYSQL_ALLOW_EMPTY_PASSWORD: false
+        ports:
+          - 3306:3306
+        options: --health-cmd="mysqladmin ping" --health-interval=10s --health-timeout=10s --health-retries=10
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+
+      # The prefix-dependencies task makes use of reflection-based PHP code that only works on PHP > 7.2.
+      - name: Install PHP 7.x for generating the vendor_prefixed directory and dependency injection
+        uses: shivammathur/setup-php@v2
+        with:
+          php-version: 7.2
+          coverage: none
+
+      - name: Install Composer dependencies, generate vendor_prefixed directory and run dependency injection
+        uses: ramsey/composer-install@v2
+
+      # Remove packages which are not PHP cross-version compatible and only used for the prefixing.
+      # - humbug/php-scoper is only needed to actually do the prefixing, so won't be shipped anyway.
+      # - league/oauth2-client and its dependencies *are* the packages being prefixed.
+      - name: Delete dependencies which are not cross-version compatible
+        run: composer remove --dev --no-scripts humbug/php-scoper league/oauth2-client
+
+      - name: Remove vendor directory
+        run: rm -rf vendor/*
+
+      - name: Install PHP
+        uses: shivammathur/setup-php@v2
+        with:
+          php-version: ${{ matrix.php_version }}
+          ini-values: zend.assertions=1, error_reporting=-1, display_errors=On
+          coverage: none
+
+      # Determine the type of Composer install which is needed.
+      # 1: WP 5.9 or higher - all PHPUnit versions needed are supported, use the most appropriate one.
+      # 2. WP < 5.9 with PHP 8.0 or higher - PHPUnit 5 - 7 supported, locked at 5.x, but needs upgrade to 7 for PHP >= 8.0.
+      # 3. WP < 5.9 with PHP < 8.0 - just use the locked PHPUnit 5 version.
+      - name: Determine the type of Composer install to use
+        id: composer_toggle
+        run: |
+          if [[ "${{ matrix.wp_version }}" =~ ^(trunk|latest|5\.9|[6789]\.[0-9])$ ]]; then
+            echo '::set-output name=TYPE::1'
+          elif [[ "${{ matrix.php_version }}" > "7.4" ]]; then
+            echo '::set-output name=TYPE::2'
+          else
+            echo '::set-output name=TYPE::3'
+          fi
+
+      - name: Debug info - show type determined
+        run: echo ${{ steps.composer_toggle.outputs.TYPE }}
+
+      # Install dependencies and handle caching in one go.
+      # Includes updating the test dependencies to the most appropriate version
+      # for the PHP/WP version combination on which the tests will be run.
+      # @link https://github.com/marketplace/actions/install-composer-dependencies
+
+      ### Install type 1.
+      - name: "Install type 1: remove PHPUnit root requirement"
+        run: composer remove --dev phpunit/phpunit --no-update --no-scripts
```

Comment:
```suggestion
        if: ${{ steps.composer_toggle.outputs.TYPE == '1' }}
        run: composer remove --dev phpunit/phpunit --no-update --no-scripts
```

Reply:
Oh cricky! Thank you. Good catch!

Reply:
Keeping my fingers crossed build will still pass now.